dashboard/views.py

```diff
@@ -290,8 +290,8 @@ def resource_access_within_week(request, course_id=0):
 
     course_date_start = get_course_date_start(course_id)
 
-    start = course_date_start + timedelta(days=(week_num_start * 7))
-    end = course_date_start + timedelta(days=(week_num_end * 7))
+    start = course_date_start + timedelta(days=((week_num_start-1) * 7))
+    end = course_date_start + timedelta(days=((week_num_end-1) * 7))
```

Comment:
```suggestion
    start = course_date_start + timedelta(days=((week_num_start - 1) * 7))
    end = course_date_start + timedelta(days=((week_num_end - 1) * 7))
```

Reply:
@jennlove-um, I think there is still an issue that we need to work through.
![Screen Shot 2021-08-04 at 11 27 26 AM](https://user-images.githubusercontent.com/35741256/128209773-b0f365af-f94e-41ea-b81c-2fefb506cad3.png)
In my opinion, resources accessed in the first week should be showing up with this setting, right? Basically, if you have "1" selected, it should show resources accessed from the first seven days; if you have "1" and "2" selected, it should show the first 14 days. I believe that's not what is happening so far here; we are only seeing the events from the weeks up to the selected number minus one.

Maybe we just need to adjust the `end` calculation so that it doesn't subtract 1?

@jonespm said this should be easy to fix, so let us know what you think.

Reply:
Yeah, currently selecting a both values on the same dot (1 to 1) only selects a specific **day**. It seems like we should either prevent this situation in the UI or treat this as a week.

1 to 1 **seems** like it's the same as saying "show me week 1" = week 1 + 7 days
1 to 2 **seems** like it's it's saying show me week 1 and week 2 (14 days)

Reply:
This is a different issue than the one identified and fixed here. The bug here was that the start and end date were both always +7 days out from where they should have been.

Reply:
I'm not sure it's strictly a different issue. With your change, 1 to 2 becomes 0 to 1, or 0-7 days. I think maybe it was correct to keep 2 as is, so it goes up to 14 days.

Reply:
Well with my change 1 to 2 becoming 0 to 1 is still 7 days, but it's the 7 days prior to week 1 rather than the 7 days after week 1. 

Do you mean 1-2 should be 0-2? That would be 21 days the way the calculation works.

Previous code, week_num_start was like 1 to 2
```
    start = course_date_start + timedelta(days=((week_num_start) * 7))
    end = course_date_start + timedelta(days=((week_num_end) * 7))
```
course_start_date is the date defined in the term or admin UI. Assume the start date is 08-01-2021 and it's weeks 1 to 2.

`start = course_start_date + 1*7 days`, so start is 08-08-2021
`end = course_start_date + 2*7 days`, end is 08-15-2021

In the code in this change
```
    start = course_date_start + timedelta(days=((week_num_start-1) * 7))
    end = course_date_start + timedelta(days=((week_num_end-1) * 7))
```

`start = course_start_date + 0*7 days`, so start is 08-01-2021
`end = course_start_date + 1*7 days`, end is 08-08-2021

Reply:
I think we're talking past each other a bit, maybe due to differing definitions we're operating under. With your start date of 8-1, I would expect "Week 1" to mean 8-1 to 8-8. It's the first week, it's "Week 1". 

In the "Weeks 1 and 2" case, if 0 and 2 were multiplied by 7, the `timedelta`s would be 0 and 14 days (not 21 days). That's the time period I would expect for selecting 1 and 2, but that needs to be agreed upon (with @jennlove-um's input).

Reply:
It all kinda boils down to whether the latter dot means to include that selected week, or go up to it. Since it's selected, I think it should include that week.